### ISSUE TITLE
Switch syscall to x/sys

### ISSUE THREAD

**Issue by clnperez** [docker/docker#31925]
An issue was opened recently for runc (https://github.com/opencontainers/runc/issues/1364) that has alerted us to the fact that syscall's TCGETS/TCSETS aren't correct for all platforms.

**Steps to reproduce the issue:**
Currently the only place we've seen this manifest itself is in the logrus test cases, which we just skipped on power since they didn't seem urgent to fix (just colors were broken).

Has there been any prior discussion on moving? It looks like the only place in the code that uses those two ioctls is in pkg/term, so that could be an initial PR.

Also a benefit of moving to sys is that you don't have to wait for new releases of go to get fixes (as with syscall).

**Describe the results you received:**
.

**Describe the results you expected:**
.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**
all versions

**Output of `docker info`:**
n/a

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by cpuguy83** [comment]
I don't think there's any good reason not to switch from the `syscall` package.
We are already importing `x/sys` as is.

**Comment by clnperez** [comment]
Thanks @cpuguy83. We'll try to squeeze this in. If anyone else has time before then, it's up for grabs. ;)

**Comment by tophj-ibm** [comment]
I've started doing this with pkg but I've ran into a problem. The stdlib syscall can be run by both windows and unix, while /x/sys needs to specifically call /x/sys/(unix|windows). There are a few spots in the codebase where the file is built on windows and unix where this is an issue https://github.com/moby/moby/blob/08d7fad45dce832dc03934e9f4fc904899220c96/distribution/utils/progress.go#L43

I've thrown this question to some of the people in the go-nuts IRC and 2 suggestions are

1.) guard the /x/sys package with buildtags (which apparently is a go-proverb https://go-proverbs.github.io/)
2.) check the GOOS before every call

IMO 1 seems like a better way to go and it looks like most calls are already separated, but I'd like to know what you think before committing to this @cpuguy83

**Comment by cpuguy83** [comment]
@tophj-ibm I don't think we need to convert *every* call into the syscall package to /x/sys.
In this case it's only a value vs functionality that can't be updated between go releases.

**Comment by binarycrusader** [comment]
Since Go 1.4, any third-party Go code should start migrating to using the sys package instead of syscall, per the official Go documentation:

> NOTE: This package is locked down. Code outside the standard Go repository should be migrated to use the corresponding package in the golang.org/x/sys repository. That is also where updates required by new systems or versions should be applied. See https://golang.org/s/go1.4-syscall for more information. 

https://golang.org/pkg/syscall/

Notably, some Linux archs such as mips64, platforms such as Solaris, and others have large sets of functionality that are missing from the syscall package.  Because of the continued use of the syscall package, any new platforms, or those that were not completely supported at the time of Go 1.4 cannot build component such as this one.

/cc @epilatow

**Comment by justincormack** [comment]
Yes use the first method.

**Comment by Xk0nSid** [comment]
Hi Is anyone working on this, I'd like to start this as my first issue?
